x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/trigger/schedule/support/TimezoneUtils.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.watcher.trigger.schedule.support;
+
+import java.time.DateTimeException;
+import java.time.ZoneId;
+import java.util.Locale;
+import java.util.Map;
+import java.util.TimeZone;
+
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * Utility class for dealing with Timezone related operations.
+ */
+public class TimezoneUtils {
```

Comment:
I debated adding this class. On the one hand is much more user friendly as they don't get a parse exception for specifying a timezone in the wrong case (With 'wrong' varying depending on the locally installed tzdb).

On the other hand it does take up additional RAM and from micro benchmarking, takes roughly 2-3x longer to parse the TZ in the worst case.

Reply:
I assume the memory usage is pretty minimal, right? And I can't imagine this taking up too much time relative to everything else that's going on?

Reply:
On the scale of things yes the memory amount is pretty small (some short strings and pointers) so I'm not too worried.

Timescales for lookups are still in the sub 50 microsecond range once the JVM is warmed up